tests/rancher-istio/rancher_istio.go

```diff
@@ -0,0 +1,947 @@
+				Set(
+					"kiali.enabled", false,
+				),
+		},
+
+		{
+			Name: "Set .Values.dns to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"dns.enabled", true,
+				),
+		},
+
+		{
+			Name: "Set .Values.dns to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"dns.enabled", false,
+				),
+		},
+
+		{
+			Name: "Set .Values.base to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"base.enabled", true,
+				),
+		},
+
+		{
+			Name: "Set .Values.base to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"base.enabled", false,
+				),
+		},
+
+		{
+			Name: "Set .Values.istiodRemote to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"istiodRemote.enabled", true,
+				),
+		},
+
+		{
+			Name: "Set .Values.istiodRemote to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"istiodRemote.enabled", false,
+				),
+		},
+
+		{
+			Name: "Set .Values.nodeSelector",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"nodeSelector", defaultNodeSelector,
+				),
+		},
+
+		{
+			Name: "Set .Values.tolerations",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				Set(
+					"tolerations", testTolerations,
+				),
+		},
+
+		{
+			Name: "Set .Values.installer repository and tag",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"installer.repository", "testRepo/mirrored-istio-installer",
+				).
+				SetValue(
+					"installer.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.installer.releaseMirror and .Values.installer.debug.secondsSleep",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"installer.releaseMirror.enabled", "true",
+				).
+				SetValue(
+					"installer.debug.secondsSleep", "30",
+				),
+		},
+
+		{
+			Name: "Set CNI to true",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.enabled", "true",
+				),
+		},
+
+		{
+			Name: "Set CNI to false",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.enabled", "false",
+				),
+		},
+
+		{
+			Name: "Set CNI repository and tag",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
+				SetValue(
+					"cni.repository", "testRepo/mirrored-istio-install-cni",
+				).
+				SetValue(
+					"cni.tag", "1.7.3",
+				),
+		},
+
+		{
+			Name: "Set .Values.cni.logLevel and .Values.cni.excludeNamespaces",
+			TemplateOptions: chart.NewTemplateOptions(DefaultReleaseName, DefaultNamespace).
```

Comment:
Merge into one `.Set()`

Reply:
done

Reply:
There still needs to be a test case for CNI set to false as well

Reply:
Never mind i see it